### DISCUSSION TITLE
Format of PETSc syntax for use in MOOSE

### DISCUSSION THREAD

**Discussion by TLWise** [idaholab/moose#27426] [Q&A General]
Given the following PETSc syntax:

-da_refine 5 -snes_monitor -ksp_monitor -snes_view -pc_type mg

How would I form the above syntax in MOOSE using "  petsc_options_iname =, and 
  petsc_options_value = " ?

**Comment by GiudGiud** [comment]
Hello

Any option that goes in pairs gets split between those two parameters (for example -pc_type lu)

Any option that stands alone (for example -ksp_view_pmat or -snes_monitor) goes in the 'petsc_options=' parameter

Or you can drop everything on the command line

Guillaume

**Comment by lindsayad** [comment]
We do not use PETSc DMDA (or DMPlex), so `-da_refine` will not do anything (you should see it show up in the options left table). Additionally we do not have support for coarsening for our DM implementation, so `-pc_type mg` will not get you anywhere. For instance if you pass `-pc_type mg -pc_mg_levels 2` to a MOOSE simulation, you will get this error
```
 0 Nonlinear |R| = 3.082207e+00
[0]PETSC ERROR: --------------------- Error Message --------------------------------------------------------------
[0]PETSC ERROR: No support for this operation for this object type
[0]PETSC ERROR: No method coarsen for DM of type libmesh
```

**Reply by lindsayad** [reply]
The most we've done in our finite element stack with respect to geometric multigrid is https://github.com/libMesh/libmesh/pull/2047
